pdns/dnsdist-tcp.cc

```diff
@@ -79,6 +79,12 @@ static std::unique_ptr<Socket> setupTCPDownstream(shared_ptr<DownstreamState>& d
         if (ds->ipBindAddrNoPort) {
           SSetsockopt(result->getHandle(), SOL_IP, IP_BIND_ADDRESS_NO_PORT, 1);
         }
+#endif
+#ifdef SO_BINDTODEVICE
+        int res = setsockopt(result->getHandle(), SOL_SOCKET, SO_BINDTODEVICE, ds->sourceItfName.c_str(), ds->sourceItfName.length());
```

Comment:
Small nit: I had to check what happens if the name is empty. You might consider doing an explicit `if (!ds->sourceItfName.empty())`...

Reply:
Oh yes, indeed!